proposals/text/0000-evaluation-haystack-2.md

```diff
@@ -0,0 +1,223 @@
+- Title: Evaluation in Haystack 2.0
+- Decision driver: (Silvano Cerza, Julian Risch)
+- Start Date: 2023-08-23
+- Proposal PR: #5794
+- Github Issue or Discussion: https://github.com/deepset-ai/haystack/issues/5265
+
+# Summary
+
+Given the below requirements we redefine how evaluation and metrics are handled in Haystack 2.x.
+
+Our goal is to lower the barrier of entry for new comers but also making it more flexible and extensible for more advanced and expert users.
+All this while making it more modular and easier to test and maintain.
+
+The requirements are:
+
+- compare the performance of different pipelines on level of pipeline outputs (user perspective, integrated eval)
+  - while running the full pipeline we can store intermediate results and calculate metrics for each component that returns answers or documents
+- find out which component is the performance bottleneck in one pipeline by evaluating subpipelines (isolated evaluation)
+- as above, get evaluation metrics for every component in a pipeline that returns or documents (ranker, retriever, reader, PromptNode)
```

Comment:
```suggestion
- as above, get evaluation metrics for every component in a pipeline that returns documents (ranker, retriever, reader, PromptNode)
```

Reply:
Do these component need to _score_ documents or to just return them? Many components return Documents but we're not interested in evaluating them (routers, preprocessors, etc...)

Reply:
Return them. This is mainly in the context of the components listed there.

The strategy here proposed doesn't make distinction of which components can be evaluated.